### PR TITLE
[Crop] add tensor-crop

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Initial commit to add tensor-crop element.
tensor-crop will crop the incoming buffer with crop info.

TODO: define pads
- 2 sink pads: static tensor for raw data, flexible for crop info
- src pad: flexible with cropped tensor

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

**Review by wooksong — COMMENTED** [review]
Q: Is it required to be a separate element rather than an extra operation in tensor transform?

**Comment by wooksong** [comment]
> @wooksong
> tensor-crop will have 2 sink pads (raw data + crop info) and 1 src pad with flexible caps (cropped tensor from raw data)

Understood.

File: gst/nnstreamer/meson.build [hunk 1]
```diff
@@ -45,6 +45,7 @@ endforeach
 nnst_plugins = [
   'tensor_aggregator',
   'tensor_converter',
+  'tensor_crop',
```
Comment (dongju-chae) [inline gst/nnstreamer/meson.build:48]: Any reason not using `tensor_transform`? Are you planning sub-plugins of `tensor-crop`?

**Review by helloahn — APPROVED** [review]
we need useful elements like this :+1:

File: gst/nnstreamer/tensor_crop/tensor_crop.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * SECTION:element-tensor_crop
+ *
+ * GStreamer element to crop the regions of incoming tensor.
+ *
+ * @file	tensor_crop.c
+ * @date	10 May 2021
+ * @brief	GStreamer element to crop the regions of incoming tensor
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Jaeyun Jung <jy1210.jung@samsung.com>
+ * @bug		No known bugs except for NYI items
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_crop/tensor_crop.h:16]: Please start the documentation section (LINE 7-9?) after Doxygen section.

File: gst/nnstreamer/tensor_crop/tensor_crop.c [hunk 3]
```diff
@@ -0,0 +1,208 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * Copyright (C) 2021 Samsung Electronics Co., Ltd.
+ */
+
+/**
+ * SECTION:element-tensor_crop
+ *
+ * GStreamer element to crop the regions of incoming tensor.
```
Comment (myungjoo) [inline gst/nnstreamer/tensor_crop/tensor_crop.c:9]: Please start the documentation section (Line 7-9) after Doxygen section. Please add an pipeline example and some descriptions as well.

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_crop/tensor_crop.c]: updated, I added todo tag for later work - element description and example pipeline.

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_crop/tensor_crop.c]: updated, I added todo tag for later work - element description and example pipeline.
After adding new PRs, I will update this section.
- define struct GstTensorCropInfo (tensor_typedef.h)
- support 2 sink pads (raw + info) in tensor_crop
- note (layout only NHWC supported)

**Review by myungjoo — APPROVED** [review]
LGTM

**Pull request merged**